R/pdf_array_job.R

```diff
@@ -71,22 +81,24 @@ pdf_array_job <- function(jobs_file, r_script, final_out_dir,
   print(command)
   system(command)
   
+  # Remove temporary directories and their contents.
+  # All have been created within this function with a UUID
+  # or explicitly passed as an argument, 
+  # so should be safe to delete.
+  temp_file_dirs = c(temp_out_dir, errors, outputs)
   if(remove_temp_files){
-    temp_dirs = c(temp_out_dir, output, errors)
-    for (i in 1:length(temp_dirs){
+    for(dir in temp_file_dirs){
       command <- paste0(
         "qsub",
         " -l fthread=2,fmem=10G,h_rt=00:00:45:00,archive=TRUE",
         " -q ", queue,
         " -P ", project,
         " -N clean_temp_files_", i
         " -now no",
-        " -hold_jid ", paste0("bind_", job_name),
+        " -hold_jid ", bind_job_name,
         " -b yes", 
-        " rm ", temp_dirs[i], "/*"
+        " rm -r ", dir[i]
```

Comment:
where is the `i` variable defined?

Reply:
What will happen here if output and errors were specified with same directory? I guess it'd create a race condition where 2 jobs try to `rm -r` the same folder? I suppose that's not a big deal

Reply:
Sorry that was a typo - it's now defined on line 89. And yes it would create a race condition, but that's probably okay.

Reply:
What if instead of `-r` the extensions were used e.g., `rm {dir[i]}/*.pdf`